src/karapace/coordinator/schema_coordinator.py

```diff
@@ -434,13 +542,20 @@ def coordinator_dead(self) -> None:
         """
         if self._coordinator_dead_fut is not None and self.coordinator_id is not None:
             LOG.warning("Marking the coordinator dead (node %s)for group %s.", self.coordinator_id, self.group_id)
+            self._are_we_master = False
             self.coordinator_id = None
             self._coordinator_dead_fut.set_result(None)
 
     def reset_generation(self) -> None:
         """Coordinator did not recognize either generation or member_id. Will
         need to re-join the group.
         """
+        LOG.info("Resetting generation status")
+        # this is called immediately after the election, we shouldn't reset this
+        # until a new node its elected aka the other path where a new node its elected
+        # otherwise this its called at each round and we keep not counting the 5 seconds
+        # required before the election.
+        # self._are_we_master = False
         self.generation = OffsetCommitRequest.DEFAULT_GENERATION_ID
```

Comment:
this was called because we were exiting the election loop in the `_async_loop` of the `master_coordinator.py`. We must keep running the thread/algorithm otherwise we are always electing a new node since that causes a rebalance and the rebalance causes a new election (the rebalance happen because we sent the `reset_generation` as a side effect of closing the heartbeat task)